packages/back-end/src/controllers/experiments.ts

```diff
@@ -2114,27 +2136,22 @@ export async function postSnapshotsWithScaledImpactAnalysis(
   const metricMap = await getMetricMap(context);
 
   // get latest snapshot for latest phase without dimensions but with results
-  let snapshotsPromises: Promise<ExperimentSnapshotInterface | null>[] = [];
-  snapshotsPromises = ids.map(async (i) => {
-    return await _getSnapshot(context, i, undefined, undefined, true);
-  });
-  const snapshots = await Promise.all(snapshotsPromises);
+  const snapshots = await _getSnapshots(context, experiments);
 
+  // Add snapshots missing scaled analysis to list to fetch
+  const experimentObjs = await getExperimentsByIds(context, experiments);
```

Comment:
You are executing this same query inside _getSnapshots already. Instead of passing in experiment IDs, can we just pass in experiment objects instead so only this outer scope is executing that query?